checkers/gameLogic.ts

```diff
@@ -948,3 +993,18 @@ export function getInitialBoard(): Board {
     ['WM', '--', 'WM', '--', 'WM', '--', 'WM', '--'],
   ];
 }
+// function isPosOnHintLine(row: number, col: number, hint: RiddleData) {
+//   switch (hint) {
+//     case 's1':
+//       return row == 0;
+//     case 's2':
+//       return row == 1;
+//     case 's3':
+//       return row == 2;
+//   }
+// }
+export function checkRiddleData(/* state: IState, turnIndex: number, firstMoveSolutions: IMove<IState>[] */): boolean {
```

Comment:
You should implement this function, in my opinion you should just call here a function with the firstMoveSolution State you have, and check if it is the winning move.
If the problem is with multiple steps for a riddle you should keep a list with the moves and when the player get to the number of moves allowed by the riddle, you should see if the move list is the same.
Should be a single function that can handle both the cases.